### PR TITLE
New edit flow

### PR DESCRIPTION
**Pull request by thomasrogerlux**
### 📝 Description

Rework the custom edit flow. For more information see the proper Slack channels.

### ❓ Context

- **Impacted projects**: `ledger-live-mobile` 
- **Linked resource(s)**: [FAT-600], [FAT-605] 

### ✅ Checklist

- [ ] ~**Test coverage**~: no UI testing on mobile yet 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo

N/A, see slack channels

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[FAT-600]: https://ledgerhq.atlassian.net/browse/FAT-600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FAT-605]: https://ledgerhq.atlassian.net/browse/FAT-605?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/pretty-terms-repair.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"live-mobile": minor
```
Comment (ofreyssinet-ledger) [inline .changeset/pretty-terms-repair.md:2]: I'd say patch, no?

File: apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+type NavigationProps = BaseComposite<
+  StackNavigatorProps<
+    CustomImageNavigatorParamList,
+    ScreenName.CustomImagePreviewPreEdit
+  >
+>;
+
+const PreviewPreEdit = ({ navigation, route }: NavigationProps) => {
+  const { t } = useTranslation();
+  const [imageToCrop, setImageToCrop] = useState<ImageFileUri | null>(null);
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx:49]: maybe rename this to `loadedImage` or something like that

File: apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+        dead = true;
+        cancel();
+      };
+    }
+    return () => {
+      dead = true;
+    };
+  }, [params, setImageToCrop, handleError]);
+
+  const [resizedImage, setResizedImage] = useState<CenteredResult | null>(null);
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx:87]: maybe rename to `croppedImage` or something like that, and uniformize with handleResizedResult etc.

File: apps/ledger-live-mobile/src/components/RootNavigator/types/CustomImageNavigator.ts [hunk 4]
```diff
@@ -13,16 +14,26 @@ type BaseParams = {
...
 export type CustomImageNavigatorParamList = {
   [ScreenName.CustomImageStep0Welcome]: BaseParams;
-  [ScreenName.CustomImageStep1Crop]: BaseParams &
-    (ImageUrl | ImageFileUri) & {
-      isPictureFromGallery?: boolean;
-    };
+  [ScreenName.CustomImageStep1Crop]: BaseParams & {
+    baseImageFile: ImageFileUri;
+    isPictureFromGallery?: boolean;
+  };
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/components/RootNavigator/types/CustomImageNavigator.ts:20]: you can remove this parameter I think

File: apps/ledger-live-mobile/src/screens/CustomImage/PreviewPostEdit.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  if (!imagePreview) {
+    return <InfiniteLoader />;
+  }
+
+  return (
+    <SafeAreaView style={{ flex: 1 }} edges={["bottom"]}>
+      <Flex flex={1}>
+        <Flex flex={1}>
+          {imagePreview?.imageBase64DataUri && (
+            <ImageProcessor
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/CustomImage/PreviewPostEdit.tsx:108]: Would it be possible to remove this processing logic completely from this step as it's already handled in the previous screen `Step2Preview` and thus we are doing the same processing twice in a row (you just need to put back the logic to get the raw result there).
In this screen then you would completely eliminate any image processing logic and get the `ProcessingPreviewResult` and `ProcessingRawResult` as route params.

File: apps/ledger-live-mobile/src/screens/CustomImage/Step2Preview.tsx [hunk 6]
```diff
@@ -200,9 +188,7 @@ const Step2Preview = ({ navigation, route }: NavigationProps) => {
           size="large"
           type="main"
           outline={false}
-          onPress={requestRawResult}
-          pending={rawResultLoading}
-          displayContentWhenPending
+          onPress={handleNavigateToPreview}
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/CustomImage/Step2Preview.tsx:191]: basically revert these change to eliminate the need to have the processing logic in PreviewPostEdit

File: apps/ledger-live-mobile/src/screens/CustomImage/Step2Preview.tsx [hunk 7]
```diff
@@ -108,25 +106,6 @@ const Step2Preview = ({ navigation, route }: NavigationProps) => {
       [setProcessorPreviewImage],
     );
 
-  const handleRawResult: ImageProcessorProps["onRawResult"] = useCallback(
```
Comment (ofreyssinet-ledger) [inline apps/ledger-live-mobile/src/screens/CustomImage/Step2Preview.tsx:111]: keep this here instead of moving it to PreviewPostEdit, to have all the processing happening in this screen only

**Review by ofreyssinet-ledger — CHANGES_REQUESTED** [review]
Great job!

Just one thing: it would be great to remove any processing logic from `PreviewPostEdit` as it's already happening in the previous screen `Step2Preview`.
It would be much simpler in my opinion. Basically you have to revert some of the changes you did in `Step2Preview` and just pass the `ProcessorRawResult` to `PreviewPostEdit`.

File: .changeset/pretty-terms-repair.md (hunk 1 above)
Comment (thomasrogerlux) [reply .changeset/pretty-terms-repair.md:2]: Why? It's not a bugfix it's the v2 of a feature

File: apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx (hunk 2 above)
Comment (thomasrogerlux) [reply apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx]: Done

File: apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx (hunk 3 above)
Comment (thomasrogerlux) [reply apps/ledger-live-mobile/src/screens/CustomImage/PreviewPreEdit.tsx]: Done

File: apps/ledger-live-mobile/src/components/RootNavigator/types/CustomImageNavigator.ts (hunk 4 above)
Comment (thomasrogerlux) [reply apps/ledger-live-mobile/src/components/RootNavigator/types/CustomImageNavigator.ts]: Done

File: apps/ledger-live-mobile/src/screens/CustomImage/PreviewPostEdit.tsx (hunk 5 above)
Comment (thomasrogerlux) [reply apps/ledger-live-mobile/src/screens/CustomImage/PreviewPostEdit.tsx]: Done

File: apps/ledger-live-mobile/src/screens/CustomImage/Step2Preview.tsx (hunk 6 above)
Comment (thomasrogerlux) [reply apps/ledger-live-mobile/src/screens/CustomImage/Step2Preview.tsx]: Done

File: apps/ledger-live-mobile/src/screens/CustomImage/Step2Preview.tsx (hunk 7 above)
Comment (thomasrogerlux) [reply apps/ledger-live-mobile/src/screens/CustomImage/Step2Preview.tsx]: Done

**Review by ofreyssinet-ledger — APPROVED** [review]
nice! good job
